firmware/application/freqman.cpp

```diff
@@ -65,61 +69,210 @@ bool load_freqman_file(std::string& file_stem, freqman_db &db) {
 		
 		// Reset line_start to beginning of buffer
 		line_start = file_data;
-		
-		if (!strstr(file_data, "f=") && !strstr(file_data, "a="))
-			break;
-		
+		 
 		// Look for complete lines in buffer
-		while ((line_end = strstr(line_start, "\x0A"))) {
-			// Read frequency
-			pos = strstr(line_start, "f=");
+		while ((line_end = strstr(line_start, "\x0A"))) {         
+
+			// Read frequency  / frequencies  		
+			//Init
+			frequency_a = 0;
 			frequency_b = 0;
-			type = SINGLE;
-			if (pos) {
-				pos += 2;
-				frequency_a = strtoll(pos, nullptr, 10);
-			} else {
-				// ...or range
-				pos = strstr(line_start, "a=");
-				if (pos) {
-					pos += 2;
-					frequency_a = strtoll(pos, nullptr, 10);
-					type = RANGE;
-					pos = strstr(line_start, "b=");
-					if (pos) {
-						pos += 2;
-						frequency_b = strtoll(pos, nullptr, 10);
-					} else
-						frequency_b = 0;
-				} else
-					frequency_a = 0;
-			}
+			type = ERROR;										// by default   third type
+			cas = 20;   // UNKNOWN (by default)
+			pos = 0 ;
+			step = STEP_DEF;
+
+			line_start[line_end - line_start] = 0 ;			// Analyze per LINE, not per Buffer 
+
+			// first test
+			if (line_end != line_start )  {    																							// test empty line
+		
+			// second test
+			if ( strstr(line_start, "f=") || strstr(line_start, "a=") || strstr(line_start, "b=") || strstr(line_start, "d=") || strstr(line_start, "//") )  {    	// no useful information 
+		
+			// third test    : no unix like line
+			if (!strstr(line_start, "\x0D")) {																							// not unix line / dos																	
+				
+			// fourth test Read comment
+			if (!strstr(line_start, "//")) { 																							// Comment we skip de line
```

Comment:
Before this functionality gets set in stone, is it worth considering using '#' as a comment delim? This is more standard than '//'